Simplenote/Classes/SPObjectManager+Simplenote.swift

```diff
@@ -49,26 +49,20 @@ extension SPObjectManager {
     }
 
     @objc
-    func newNote(from urlComponents: URLComponents) -> Note {
-        guard let queryItems = urlComponents.queryItems else {
-            return newDefaultNote()
-        }
-
+    func newNote(withContent content: String?, tags: [String]?) -> Note {
         let newNote = newDefaultNote()
 
-        for item in queryItems {
-            if item.name == "content" {
-                newNote.content = item.value
-            } else if item.name == "tag" {
-                if let tags = item.value?.components(separatedBy: " ") {
-                    for tag in tags {
-                        if tag.count == 0 {
-                            continue
-                        }
-                        newNote.addTag(tag)
-                        SPObjectManager.shared().createTag(from: tag)
-                    }
+        if let content = content {
+            newNote.content = content
+        }
+
+        if let tags = tags {
+            for tag in tags {
+                if tag.isEmpty {
+                    continue
                 }
+                newNote.addTag(tag)
+                SPObjectManager.shared().createTag(from: tag)
```

Comment:
I know that it wasn't here, but to be consistent we should apply the same validation/preparation rules that are used when adding tags via UI.

I think this should do the trick

```
let validator = TagTextFieldInputValidator()
tags?
    .compactMap({ validator.preprocessForPasting(tag: $0) })
    .forEach { tag in
        newNote.addTag(tag)
        SPObjectManager.shared().createTag(from: tag)
    }
```